src/inmanta/data/__init__.py

```diff
@@ -530,10 +404,17 @@ def get_order(self) -> PagingOrder:
         pass
 
     @abstractmethod
-    def get_paging_boundaries(self, first: dict[str, object], last: dict[str, object]) -> PagingBoundaries:
+    def get_paging_boundaries(self, first: abc.Mapping[str, object], last: abc.Mapping[str, object]) -> PagingBoundaries:
         """Return the page boundaries, given the first and last record of the page"""
         pass
 
+    @abstractmethod
+    def is_id_nullable(
+        self,
+    ) -> bool:
+        """Is the (secondary) order column nullable?"""
+        pass
+
```

Comment:
I don't really get the meaning of this yet. Might come to me when I've seen more of the code.

Reply:
Could it be that you made a naming error here (and in `RequestedPagingBoundaries`)? From how it's used in `RequestedPagingBoundaries.validate` I get the impression that it's about the primary column being nullable, i.e. the secondary value being required for valid filtering and sorting.